README.md

```diff
@@ -83,6 +83,50 @@ fastify.get('/', (request, reply) => {
 fastify.listen(3000)
 ```
 
+## Using Keys with key rotation
+
+It's possible to use an array for the key field to support key rotation as an additional security measure.
+Cookies will always be signed with the first key in the array to try to "err on the side of performance" however
+if decoding the key fails, it will attempt to decode using every subsequent value in the key array.
+
+WARNING: This operation can get expensive if too many keys are used. It is recommended to only use 2 keys at anytime.
+This should allow ample support time for supporting sessions with an old key while rotating to the new one.
+
+IMPORTANT: The new key you are trying to rotate to should always be the first key in the array.
```

Comment:
This needs to be explained a bit more in detail. By using 2 or 3 keys, whenever a value is to the cookie it would be reenconded with the new key. Or am I wrong?

Reply:
Yes as it's currently written it doesn't reencode with the first key but thinking about it, that is probably a better thing to do for performance so it doesn't keep needing to go to the second key every time for that user.

Reply:
I pushed up an update so that it would re-sign the session with the latest key to keep performance high. I also added additional explanation and a reference to the test case that hits this functionality. Let me know if there's a different direction we should take it.